sources/packages/backend/apps/api/src/services/student-assessment/student-assessment.service.ts

```diff
@@ -236,4 +246,97 @@ export class StudentAssessmentService extends RecordDataModelService<StudentAsse
       .addOrderBy("assessment.submittedDate", "DESC")
       .getMany();
   }
+
+  /**
+   * Creates a manual assessment for the application.
+   * @param applicationId application id.
+   * @param userId user id who triggered the manual reassessment.
+   */
+  manualReassessment(applicationId: number, note: string, userId: number) {
+    return this.dataSource.transaction(async (transactionalEntityManager) => {
+      const currentUser = { id: userId } as User;
+      const now = new Date();
+      const applicationRepo =
+        transactionalEntityManager.getRepository(Application);
+      const application = await applicationRepo.findOne({
+        select: {
+          id: true,
+          currentAssessment: true as unknown,
+          student: { id: true },
+        },
+        relations: [
+          "currentAssessment",
+          "currentAssessment.offering",
+          "currentAssessment.studentAppeal",
+          "student",
+        ],
+        where: {
+          id: applicationId,
+        },
+      });
+
+      if (!application) {
+        throw new CustomNamedError(
+          `Application not found.`,
+          APPLICATION_NOT_FOUND,
+        );
+      }
+      if (
+        application.currentAssessment.studentAssessmentStatus !==
+        StudentAssessmentStatus.Completed
+      ) {
+        throw new CustomNamedError(
+          `Application current assessment expected to be '${StudentAssessmentStatus.Completed}' to allow manual reassessment.`,
+          INVALID_OPERATION_IN_THE_CURRENT_STATUS,
+        );
+      }
+      if (application.isArchived) {
+        throw new CustomNamedError(
+          `Application cannot have manual reassessment after being archived.`,
+          INVALID_OPERATION_IN_THE_CURRENT_STATUS,
+        );
+      }
+
+      // Adds application note.
+      const newNote = new Note();
+      newNote.description = note;
+      newNote.noteType = NoteType.Application;
+      newNote.creator = currentUser;
+      newNote.createdAt = now;
+      const savedNote = await transactionalEntityManager
+        .getRepository(Note)
+        .save(newNote);
+      // Associates the created note with the student.
+      await transactionalEntityManager
+        .getRepository(Student)
+        .createQueryBuilder()
+        .relation(Student, "notes")
+        .of(application.student)
+        .add(savedNote);
+
+      const oldCurrentAssessment = application.currentAssessment;
+      const newCurrentAssessment = new StudentAssessment();
```

Comment:
Just a suggestion. Maybe we could name the two variables as `originalAssessment` and `manualReassessment` here?

Reply:
I wouldn't like to call it `originalAssessement` to not get confused with the triggerType `originalAssessement`. The old assessment can be a `manualReassessment` too. I think the var names for this context are good.